### PR TITLE
Reduce Lambda function cold start from 11 seconds to less than 2 seconds

### PR DESCRIPTION
**Pull request by dmahapatro**
**Description of changes:**

Existing implementation of Lambda functions had below problems:
- Used Java SDK v1 which is less performant than v2
- Targeted for JDK 8 Lambda runtime which is old
- Had a cold start of more than 11 seconds

With this PR below items are taken care of:
- Upgrade Gradle to 7.1 and use Gradle wrapper
- Upgrade to use AWS Java SDK v2
- Upgrade app to use `DynamoDbBean` and `DynamoDbEnhancedAsyncClient`
- Use AWS CRT http client for better performance
- Upgrade dependencies to latest as of 3/19/2022
- Upgrade to use Java 11 Corretto instead of Java 1.8
- Added .editorconfig for consistent formatting
- Removed unused classes and code
- Implement Java Lambda best practices

**Result:**
Lambda function cold start reduced from 11 seconds to below 2 seconds (~1.7s for GetUnicornBasket API with multiple unicorns)

**Configuration**
- Runtime: Java 11
- Memory: 512MB
- Architecture: arm64

**How to create Lambda artifact?**
Use below command to create the artifact `MonoToMicroLambda-0.0.1.jar`:

```bash
./gradlew clean shadowJar
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by deki** [comment]
Nice work, this was on the backlog for a long time :-)

We need to adapt the instructions https://catalog.us-east-1.prod.workshops.aws/workshops/f2c0706c-7192-495f-853c-fd3341db265a/en-US/microservices and can merge it afterwards.

**Comment by dmahapatro** [comment]
Thanks @deki . I have been talking to Sabha Parameswaran about this. I do not have access to workshop studio. This workshop is planned for an XXL GFS customer and we need to fix the content in the workshop too.

Will you be able to provide me or Sabha access to workshop studio? We need to fix that asap.

**Comment by msailes** [comment]
Can I suggest we add the tiered compilation stop at level 1 config as well. 

Example:
https://github.com/aws-samples/aws-lambda-java-tiered-compilation-example/blob/46cb440fe32b285b2abd8ad77c5fac10d458b1f3/infrastructure/src/main/java/example/InfrastructureStack.java#L70

**Comment by dmahapatro** [comment]
The workshop uses manual creation of a Lambda function via AWS Console. Most likely this has to be a step in the workshop guide.

**Comment by dmahapatro** [comment]
Tested adding tiered compilation stop at level 1 config. Init duration (cold start) reduced from 1769ms to 1560ms @msailes

File: MonoToMicroAssets/MonoToMicroCF.template [hunk 1]
```diff
@@ -367,7 +367,7 @@
                     "configure" : {
                         "commands" : {
                             "0-clone-repo" : { "command" :
-                                "git clone https://github.com/aws-samples/unishop-monolith-to-microservices.git /home/ec2-user/MonoToMicro"
+                                "git clone https://github.com/dmahapatro/unishop-monolith-to-microservices.git /home/ec2-user/MonoToMicro"
```
Comment (dmahapatro) [inline MonoToMicroAssets/MonoToMicroCF.template]: @deki Assume this is just for testing purposes?

Comment (deki) [reply MonoToMicroAssets/MonoToMicroCF.template]: yes, will fix it before merging

**Pull request merged**